### DISCUSSION TITLE
How to use simple file inputs

### DISCUSSION THREAD

**Discussion by ilikali** [refinedev/refine#1654] [Q&A] [answered]
Hi. I read in the documentation about file uploads (https://refine.dev/docs/guides-and-concepts/upload/multipart-upload/), but I would like to know how you can do a simple file upload using type="file".  My server takes one value for the image. 

How can I make an implementation of this code? 

`var formdata = new FormData();`
`formdata.append("preview", fileInput.files[0], "image.jpeg");`

Thanks

**Comment by omeraplak** [comment]
Hey @ilikali ,
Thank you for contacting us! Are you using Ant Design or custom UI?

**Reply by ilikali** [reply]
Custom UI )

`<input {...register("preview")} type="file"/>`

**Reply by omeraplak** [reply]
I found an example for you. Does it work for you?

https://codesandbox.io/s/fancy-breeze-f7gov?fontsize=14&hidenavigation=1&theme=dark&file=/src/App.js

**Comment by mlhekinci** [comment, marked as answer]
Hey @ilikali, we recently wrote a blog post on this topic and included it in our documentation.

[Refine React Hook Form Docs](https://refine.dev/docs/packages/react-hook-form/#multipart-file-upload)

[Refine Multipart File Upload Blog](https://refine.dev/blog/how-to-multipart-file-upload-with-react-hook-form/#how-to-multipart-file-upload-with-refine-and-react-hook-form)

[CodeSandbox](https://codesandbox.io/s/refine-react-hook-form-example-4hf74?from-embed)

By following the document and examples, you can get an idea of ​​how to upload files to your server.

I hope it helps.

**Reply by ilikali** [reply]
I used a very similar approach. Anyway, you are great and thanks for developing the refine!
